### PR TITLE
Updates logr, zapr, packet logr, bmclib dependencies

### PR DESCRIPTION
**Pull request by joelrebel**
## Description
logr, zapr, packetlogr, bmclib dependencies updated

github.com/go-logr/logr v0.4.0 -> v1.2.2
github.com/go-logr/zapr v0.4.0 -> v1.2.2
github.com/bmc-toolbox/bmclib v0.4.14 -> v0.4.16
packethost/pkg/log/logr -> packethost/pkg/log/logr/v2

bmclib updated in this change because of the indirect (bmclib v0.4.14) dependency on the on logr v0.4.0.

note: 
expects linked PR merged https://github.com/packethost/pkg/pull/51 and tagged with `log/logr/v2.0.0`  
once the linked PR is approved, merged and tagged, go.mod is to be updated to remove the `replace` directive
https://github.com/tinkerbell/pbnj/pull/113/commits/2a4ece95ff20770e53c9df1be9e1e5700e90d56f#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R44

## Why is this needed

The current logr, zapr, packethost/pkg/logr depend on the older 'beta' `logr` package,
this change updates the logging dependencies to the current latest.

for more information see https://github.com/packethost/pkg/pull/51

## How are existing users impacted? What migration steps/scripts do we need?

No impact on existing users identified

### CODE REVIEW

File: grpc/server.go [hunk 1]
```diff
@@ -114,18 +114,18 @@ func RunServer(ctx context.Context, log logging.Logger, grpcServer *grpc.Server,
...
 
 	go func() {
 		<-ctx.Done()
-		log.Info("ctx cancelled, shutting down PBnJ")
+		log.Info(0, "ctx cancelled, shutting down PBnJ")
 		grpcServer.GracefulStop()
 	}()
 
-	log.Info("starting PBnJ gRPC server")
+	log.Info(0, "starting PBnJ gRPC server")
```
Comment (jacobweinstock) [inline grpc/server.go:129]: Hey @joelrebel, thanks for tackling this. I'm concerned that this logging signature now doesn't follow the go-logr documentation `log.V(0).Info`. I don't think this allows us to use `go-logr` as is expected/intended. I think this is only necessary because of the way logging is being handled in general here in PBnJ. It's not very good 😞  (I wrote it!). 

What are your thoughts around removing the`packethost/pkg/log/logr` dependency entirely and just using `go-logr`? I apologize. I know it's not what you were wanting to do here and the work to move to `go-logr` touches a lot. I'm willing to do the work to do this update if that's helpful.

**Comment by abhay-krishna** [comment]
Hi PBNJ maintainers, we build and distribute an opinionated version of [PBNJ](https://github.com/aws/eks-anywhere-build-tooling/tree/main/projects/tinkerbell/pbnj) as part of the EKS Anywhere artifact catalog. PBNJ has a go mod dependency on [bmc-toolbox/bmclib](https://github.com/bmc-toolbox/bmclib), which had a dependency on the LGPLv3-licensed [gebn/bmc](https://github.com/gebn/bmc). This was fixed in bmc-toolbox/bmclib#257, and the intention was to make our build of PBNJ use this newer commit in its Go mod through a patch.

However, (as you all know) bmclib has since [upgraded logging dependencies](https://github.com/bmc-toolbox/bmclib/commit/3e94ce7ca5169786361fc8aed7d22ca267415d41), so directly consuming this commit was causing incompatibility issues with some other Go mods such as [packethost/pkg/log/logr](https://github.com/packethost/pkg/tree/master/log/logr) which are using much older versions of `logr` and `zapr` modules.

In order to get around this, we follow the approach of using a [local bmclib module](https://github.com/aws/eks-anywhere-build-tooling/blob/main/projects/tinkerbell/pbnj/patches/0001-Replace-bmclib-mod-with-local-relative-path-for-LGPL.patch) version of bmclib which has the [PR's commit cherry-picked](https://github.com/aws/eks-anywhere-build-tooling/blob/main/projects/tinkerbell/pbnj/patches/bmclib/0001-Remove-goipmi-provider-from-bmclib.patch).

# TLDR
**Would really love for this PR to be merged!** 🎉

File: grpc/server.go (hunk 1 above)
Comment (jacobweinstock) [reply grpc/server.go:129]: Yeah, if you wouldn't mind closing this one. I opened https://github.com/tinkerbell/pbnj/pull/116 to do the move to `go-logr` directly.

**Comment by jacobweinstock** [comment]
> Hello PBNJ maintainers, we build and distribute an opinionated version of [PBNJ](https://github.com/aws/eks-anywhere-build-tooling/tree/main/projects/tinkerbell/pbnj) as part of the EKS Anywhere artifact catalog. PBNJ has a go mod dependency on [bmc-toolbox/bmclib](https://github.com/bmc-toolbox/bmclib), which had a dependency on the LGPLv3-licensed [gebn/bmc](https://github.com/gebn/bmc). This was fixed in [bmc-toolbox/bmclib#257](https://github.com/bmc-toolbox/bmclib/pull/257), and the intention was to make our build of PBNJ use this newer commit in its Go mod through a patch.
> 
> However, (as you all know) bmclib has since [upgraded logging dependencies](https://github.com/bmc-toolbox/bmclib/commit/3e94ce7ca5169786361fc8aed7d22ca267415d41), so directly consuming this commit was causing incompatibility issues with some other Go mods such as [packethost/pkg/log/logr](https://github.com/packethost/pkg/tree/master/log/logr) which are using much older versions of `logr` and `zapr` modules.
> 
> In order to get around this, we follow the approach of using a [local bmclib module](https://github.com/aws/eks-anywhere-build-tooling/blob/main/projects/tinkerbell/pbnj/patches/0001-Replace-bmclib-mod-with-local-relative-path-for-LGPL.patch) version of bmclib which has the [PR's commit cherry-picked](https://github.com/aws/eks-anywhere-build-tooling/blob/main/projects/tinkerbell/pbnj/patches/bmclib/0001-Remove-goipmi-provider-from-bmclib.patch).
> 
> # TLDR
> **Would really love for this fix to be merged! (or moving to go-logr)** 🎉

Hey @abhay-krishna. This new PR https://github.com/tinkerbell/pbnj/pull/116 should hopefully satisfy this. Mind checking it out?

**Comment by abhay-krishna** [comment]
Thank you! I'll take a look asap!
